osconfig_tests/test_suites/patch/patch.go

```diff
@@ -0,0 +1,186 @@
+	"context"
+	"fmt"
+	"github.com/GoogleCloudPlatform/compute-image-tools/osconfig_tests/osconfig_server"
+	"github.com/GoogleCloudPlatform/compute-image-tools/osconfig_tests/utils"
+	"github.com/golang/protobuf/ptypes/duration"
+	"github.com/kylelemons/godebug/pretty"
+	"log"
+	"path"
+	"regexp"
+	"sync"
+	"time"
+
+	osconfigpb "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/google-osconfig-agent/_internal/gapi-cloud-osconfig-go/google.golang.org/genproto/googleapis/cloud/osconfig/v1alpha1"
+	daisyCompute "github.com/GoogleCloudPlatform/compute-image-tools/daisy/compute"
+	"github.com/GoogleCloudPlatform/compute-image-tools/osconfig_tests/compute"
+	"github.com/GoogleCloudPlatform/compute-image-tools/osconfig_tests/junitxml"
+	api "google.golang.org/api/compute/v1"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/osconfig_tests/test_config"
+)
+
+const (
+	testSuiteName = "PatchTests"
+)
+
+var (
+	dump = &pretty.Config{IncludeUnexported: true}
+)
+
+var vf = func(inst *compute.Instance, vfString string, port int64, interval, timeout time.Duration) error {
+	return inst.WaitForSerialOutput(vfString, port, interval, timeout)
+}
+
+type patchTestSetup struct {
+	image         string
+	name          string
+	startup       *api.MetadataItems
+	vstring       string
+	assertTimeout time.Duration
+	vf            func(*compute.Instance, string, int64, time.Duration, time.Duration) error
+}
+
+// TestSuite is a PatchTests test suite.
+func TestSuite(ctx context.Context, tswg *sync.WaitGroup, testSuites chan *junitxml.TestSuite, logger *log.Logger, testSuiteRegex, testCaseRegex *regexp.Regexp, testProjectConfig *testconfig.Project) {
+	defer tswg.Done()
+
+	if testSuiteRegex != nil && !testSuiteRegex.MatchString(testSuiteName) {
+		return
+	}
+
+	testSuite := junitxml.NewTestSuite(testSuiteName)
+	defer testSuite.Finish(testSuites)
+
+	logger.Printf("Running TestSuite %q", testSuite.Name)
+
+	suffix := utils.RandString(5)
+	image := "projects/debian-cloud/global/images/family/debian-9"
+
+	testSetup := []*patchTestSetup{
+		// Windows images.
+		&patchTestSetup{
+			image:   image,
+			name:    fmt.Sprintf("patch-test-%s-%s", path.Base(image), suffix),
+			vstring: "Finished patch job",
+			startup: &api.MetadataItems{
+				Key:   "startup-script",
+				Value: &utils.InstallOSConfigDeb,
+			},
+			vf:            vf,
+			assertTimeout: 1200 * time.Second,
+		},
+	}
+
+	var wg sync.WaitGroup
+	tests := make(chan *junitxml.TestCase)
+	for _, setup := range testSetup {
+		wg.Add(1)
+		go patchTestCase(ctx, setup, tests, &wg, logger, testCaseRegex, testProjectConfig)
+	}
+
+	go func() {
+		wg.Wait()
+		close(tests)
+	}()
+
+	for ret := range tests {
+		testSuite.TestCase = append(testSuite.TestCase, ret)
+	}
+
+	logger.Printf("Finished TestSuite %q", testSuite.Name)
+}
+
+func runCreatePatchTest(ctx context.Context, testCase *junitxml.TestCase, testSetup *patchTestSetup, logger *log.Logger, testProjectConfig *testconfig.Project) {
```

Comment:
"ExecutePatchTest". There is no "CreatePatch" function.

Reply:
done